### PR TITLE
System test

### PR DESCRIPTION
**Pull request by biruk-bereka**
# In this PR We:
   - Fix N+1 problem when fetching all posts and their comments for a user.
   ## Before Fixing N + 1 problem
before-1
before-2
before-3
  ## After Fixing N + 1 problem
after-1
after-2

   - Use Capybara to write integration tests for each view in a project.
   - Write integration test for user index and show views
   - Write integration test for post index and show views

I did this project with my partner @GhislainMitahi.

**NB: run `rails tailwindcss:watch` to build the css for the project.**

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -20,6 +20,6 @@ def update_post_counter
 
   # A method that returns the 5 most recent comments for a given post
   def recent_comments
-    Comment.where('post_id = ?', id).order('created_at ASC').limit(5)
+    Comment.includes(:author).where('post_id = ?', id).order('created_at ASC').limit(5)
   end
```
Comment (tresorsawasawa) [inline app/models/post.rb:24]: - [ ] As required for the project, kindly make sure you We ask students you add screenshots of the logs **Before** and **After** the fix to the PR description.

File: spec/integration/posts/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  it 'should have section for pagination if there are more posts than fit on the view' do
+    visit user_posts_path(user)
+    expect(page).to have_button('Next')
+  end
+
+  it 'redirect to user\'s posts page when click on the post\'s title' do
+    visit user_posts_path(user)
+    click_on(p0.title)
+    expect(page).to have_current_path(user_post_path(user_id: p0.author_id, id: p0.id))
+  end
```
Comment (tresorsawasawa) [inline spec/integration/posts/post_index_spec.rb]: - [ ] As required for the project, kindly make sure your `User` **post index page** has the following featured test(s):
  * Tests that asserts you can see the first comments on a post.

File: spec/integration/posts/post_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  it 'should show the username of the comment\'s author' do
+    visit user_post_path(post)
+    expect(page).to have_content('Jerry')
+  end
+
+  it 'should show the comment\'s text' do
+    visit user_post_path(post)
+    expect(page).to have_content(comment.text)
+  end
```
Comment (tresorsawasawa) [inline spec/integration/posts/post_show_spec.rb:56]: - [ ] As required for the project, kindly make sure your `User` **post index page** has the following featured test(s):
  * Tests that asserts you can see the comment each commentor left.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @biruk-bereka and @GhislainMitahi 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- All integration tests are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @biruk-bereka and @GhislainMitahi,

✔️ Well written and passing tests

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**